File: less2/h/main.py
import sys


class Bank:
    def __init__(self):
        self.accounts = {}

    def DEPOSIT(self, name: str, sum_: int) -> None:
        self.accounts[name] = self.accounts.get(name, 0) + sum_

    def WITHDRAW(self, name: str, sum_: int) -> None:
        self.accounts[name] = self.accounts.get(name, 0) - sum_

    def BALANCE(self, name: str):
        return self.accounts.get(name)

    def TRANSFER(self, name1: str, name2: str, sum_: int) -> None:
        self.accounts[name1] = self.accounts.get(name1, 0) - sum_
        self.accounts[name2] = self.accounts.get(name2, 0) + sum_

    def INCOME(self, p: int) -> None:
        for name, balance in list(self.accounts.items()):
            if balance > 0:
                self.accounts[name] = balance * (100 + p) // 100


def main():
    b = Bank()
    for line in sys.stdin:
        parts = line.split()
        op = parts[0]

        if op == "DEPOSIT":
            name = parts[1]
            sum_ = int(parts[2])
            b.DEPOSIT(name, sum_)
        elif op == "WITHDRAW":
            name = parts[1]
            sum_ = int(parts[2])
            b.WITHDRAW(name, sum_)
        elif op == "BALANCE":
            name = parts[1]
            sum_ = b.BALANCE(name)
            if sum_ is not None:
                print(sum_)
            else:
                print("ERROR")
        elif op == "TRANSFER":
            name1 = parts[1]
            name2 = parts[2]
            sum_ = int(parts[3])
            b.TRANSFER(name1, name2, sum_)
        elif op == "INCOME":
            p = int(parts[1])
            b.INCOME(p)
        else:
            raise RuntimeError(f"unknown operation: {op}")


if __name__ == "__main__":
    main()

File: less2/h/main.go
package main

import (
	"errors"
	"io"
	"os"
	"strconv"
	"strings"

	. "github.com/aaa2ppp/contestio"
)

var debug bool

type Bank struct {
	accounts map[string]int
}

func NewBank() *Bank {
	return &Bank{make(map[string]int)}
}

func (b *Bank) DEPOSIT(name string, sum int) {
	b.accounts[name] += sum
}

func (b *Bank) WITHDRAW(name string, sum int) {
	b.accounts[name] -= sum
}

func (b *Bank) BALANCE(name string) (int, bool) {
	sum, exists := b.accounts[name]
	return sum, exists
}

func (b *Bank) TRANSFER(name1, name2 string, sum int) {
	b.accounts[name1] -= sum
	b.accounts[name2] += sum
}

func (b *Bank) INCOME(p int) {
	for customer, sum := range b.accounts {
		if sum > 0 {
			sum = sum * (100 + p) / 100
			b.accounts[customer] = sum
		}
	}
}

type solveFunc func()

func solve() {}

func run(in io.Reader, out io.Writer, solve solveFunc) {
	br := NewReader(in)
	bw := NewWriter(out)
	defer bw.Flush()

	b := NewBank()

	var err error
	var op string
loop:
	for {
		op, err = br.ReadString('\n')
		if err != nil {
			break
		}
		f := strings.Fields(op)
		switch op := f[0]; op {
		case "DEPOSIT":
			name := f[1]
			sum, _ := strconv.Atoi(f[2])
			b.DEPOSIT(name, sum)
		case "WITHDRAW":
			name := f[1]
			sum, _ := strconv.Atoi(f[2])
			b.WITHDRAW(name, sum)
		case "BALANCE":
			name := f[1]
			sum, exists := b.BALANCE(name)
			if exists {
				PrintIntLn(bw, sum)
			} else {
				PrintWordLn(bw, "ERROR")
			}
		case "TRANSFER":
			name1 := f[1]
			name2 := f[2]
			sum, _ := strconv.Atoi(f[3])
			b.TRANSFER(name1, name2, sum)
		case "INCOME":
			p, _ := strconv.Atoi(f[1])
			b.INCOME(p)
		default:
			err = errors.New("unknown operation: " + op)
			break loop
		}
	}
}

func main() {
	run(os.Stdin, os.Stdout, solve)
}
